 #### 1. Take the input from the user for(Total number of people,Number of seats for bus. Based on two inputs dDecide how many number of buses required.
   


In [ ]:
total_people = int(input("Enter the total_number of people:" ))
seats_per_bus = int(input("Enter the number of seats per bus: " ))
if (total_people % seats_per_bus)==0:
    no_of_buses_required = total_people // seats_per_bus
else:
    no_of_buses_required = (total_people // seats_per_bus)+1
    
print(no_of_buses_required)

#### 2.Take temperature from the user and convert foreign heat -> Celsius.

In [ ]:
fahrenheit = 98.6  
celsius = (fahrenheit - 32) * 5 / 9
print("Temperature in Celsius:", celsius)



#### 3. Take temperature from the user and convert Celsius → foreign heat.


In [ ]:
celsius = 78.6  
fahrenheit = (fahrenheit + 32) * 5 / 9
print("Temperature in fahrenheit:", fahrenheit)


#### 4. Take four number from the user (variables name it as x1,x2,x3,x4)
#### Do the below operations
 #### (x1+x2)**2, (x3+x4)**3
#### variance
#### standard deviation: sqrt(variance):  User math module. Math.sqrt(variance)
#### Regression
	## y=mx+b
          m=1.23
          b=0.045
          find out y
          y=m*(x1+x2+x3+x4)+b
#### Find the average of four numbers
#### Find the sum of four numbers


In [ ]:

import math 
x1 = int(input("Enter the value of x1: "))
x2 = int(input("Enter the value of x2: "))
x3 = int(input("Enter the value of x3: "))
x4 = int(input("Enter the value of x4: "))

result1 = (x1 + x2) ** 2
result2 = (x3 + x4) ** 3

print("The result of (x1 + x2)^2 is:", result1)
print("The result of (x3 + x4)^3 is:", result2)
 
mean = (x1 + x2+ x3 + x4 )//4 
print(mean)

numbers = (x1 , x2, x3 , x4 )
sum_sqd_numbers = 0
sqd_numbers = []
for i in numbers:
    sqd = (i - mean) ** 2
    sum_sqd_numbers += sqd
    sqd_numbers.append(sqd)
variance = sum_sqd_numbers/(4-1)
print(" variance : ",variance)    

std_dev = math.sqrt(variance)
#print(std_dev)
## y=mx+b
m=1.23
b=0.045
y  =  m * (x1+x2+x3+x4) +b
print("regression : ", y)
print("standard deviationRegression : ",std_dev)
print("average of four numbers : ",mean)
print("sum of 4 numbers : ",(x1+x2+x3+x4))



#### 5. Take the distance in km.Show that in cm, meters, in milli meters, cents, feets, yards


In [ ]:
km = float(input("Enter the distance in kilometers: "))
meters = km * 1000        
centimeters = km * 100000 
millimeters = km * 1e6    
feet = km * 3280.84     
yards = km * 1093.61 
print(f"\nDistance in different units:")
print(f"{km} kilometers = {centimeters} centimeters")
print(f"{km} kilometers = {meters} meters")
print(f"{km} kilometers = {millimeters} millimeters")
print(f"{km} kilometers = {feet} feet")
print(f"{km} kilometers = {yards} yards")

#### 6.Take the size of your hard disk in GB.Show that in MB, KB, TB, PB


In [ ]:
gb = float(input("Enter the size of the hard disk in GB: "))
mb = gb * 1024           
kb = gb * 1024 * 1024    
tb = gb / 1024           
pb = gb / (1024 * 1024)
print(f"\nSize in different units:")
print(f"{gb} GB = {mb} MB")
print(f"{gb} GB = {kb} KB")
print(f"{gb} GB = {tb} TB")
print(f"{gb} GB = {pb} PB")

#### 7. Take name, age, height from the user and print like below.The details of the person: Name:name of the person, Age:age of the person, Height:height of the person


In [ ]:
name = input("Enter your name: ")
age = int(input("Enter your age: "))
height = float(input("Enter your height in meters: "))
print(f"The details of the: Name:{name}, Age:{age}, Height:{height}")


#### 8. BMI calculation: take required parameters for BMI calculation from the user and calculate BMI of the person.

In [ ]:
weight = float(input("Enter your weight in kilograms: "))
height = float(input("Enter your height in meters: "))
bmi = weight / (height ** 2)
print(f"Your BMI is: {bmi:.2f}")


#### 9. name="Jayaram"
 age=1.6
 height=3.5356234
 weight=10.343856783
 By using above inputs print the output
 Name:Jayaram, Age:1.6, Height:3.54, Weight:10.344


In [ ]:
name = "Jayaram"
age = 1.6
height = 3.5356234
weight = 10.343856783
print("Name:%s, Age:%.1f, Height:%.2f, Weight:%.3f" % (name, age, height, weight))


####  10.Take three upper case letters from the user convert in to small case.

In [ ]:
letter1 = input("Enter the first uppercase letter: ")
letter2 = input("Enter the second uppercase letter: ")
letter3 = input("Enter the third uppercase letter: ")
smallercase_letter1 = letter1.lower()
smallercase_letter2 = letter2.lower()
smallercase_letter3 = letter3.lower()
print(f"smallercase letters: {smallercase_letter1}, {smallercase_letter2}, {smallercase_letter3}")


#### 11. Take base and exponent value from the user and print like in mathematics: example: base=2, exponent=3: 23


In [ ]:
# Taking base and exponent as input
base = int(input("Enter the base value: "))
exponent = int(input("Enter the exponent value: "))
if exponent == 2:
    print(f"{base}\u00b2")  
elif exponent == 3:
    print(f"{base}\u00b3")  
else:
    print(f"{base}^{exponent}")  



#### 12. Take some groceries cost prices and print total cost and average cost, what is the max cost, what is the minimum cost.

In [ ]:
grocery_costs = [50, 30, 40, 60, 25]
total_cost = sum(grocery_costs)
average_cost = total_cost / len(grocery_costs)
max_cost = max(grocery_costs)
min_cost = min(grocery_costs)
print("Total_Cost:", total_cost)
print("Average_Cost:", average_cost)
print("Maximum_Cost:", max_cost)
print("Minimum_Cost:", min_cost)

#### 13.Take the input from the user for(Total number of people, total number of buses, Number of seats for bus, adjust factor). Based on four input Decide whether there is sufficient buses or not and give solution for how many extra buses required.


                    

In [ ]:
totalpeople = int(input("Enter total no of people:"))
totalbuses = int(input("Enter total no of buses:"))
seatsforbus = int(input("Enter total no of seats for bus:"))
adjustfactor = int(input("Enter adjust factor:"))
totalseats = totalbuses*seatsforbus*adjustfactor
if totalseats >= totalpeople:
   print("sufficient buses are available:")
else:
     extrapeople = totalpeople - totalseats
     extrabuses = (extrapeople + (seatsforbus - 1))
     print(f"sufficinet buses are not available.You need more buses.")
    

#### 14.Take number from the user decide whether it is even or odd.

In [ ]:
num = int(input("Enter the Number:"))
if num % 2 == 0:
    print("Number is even:")
else:
    print("Number is odd:")

#### 15.Take number from the user decide whether it is positive number or negative number.

In [ ]:
num = int(input("Enter the Number:"))
if num > 0:
    print("Number is positive:")
elif num < 0:
    print("Number is Negative:")
else:
    print("Number is Zero:")

#### 16.Take a string from the user print the length. if the user not given anything then show an error message

In [ ]:
userinput = input("Enter a string:")
if userinput:
    print("The length is:({userinput})")
else:
    print("Error")

#### 17.Take a string from the user print the length. if the user not given anything then show an error message

In [ ]:
user_input = input("Enter a string:")

if user_input:
        print("Length of the string:",len(user_input))
else:
        print("Error:")
         

        


#### 18.code to perform mathematical operations. take two numbers from the user and show the below menu
	1. add,
	2. sub,	
	3. mul,
 	4.div, 
	5.quit
	Enter an option:
		based on the option need to perform an operations


In [ ]:
num1=int(input("Enter First number:"))
num2=int(input("Enter second number:"))
options=int(input("1 for add\n 2 for sub\n 3 for mul\n 4 for div\n 5 for quit\n:"))
if options==1:
    print(num1+num2)
if options==2:
    print(num1-num2)
if options==3:
    print(num1*num2)
if options==4:
    print(num1/num2)
if options==5:
    print("quit")
    

#### 22. Take an age  and gender from the user: and mention that what he/she can 	do in india.
	"""
		conditions
1. Theatre: 5 for men 7 for women
    	2. Voting system: 18 for men and women
    	3. Marriage in india: 23 for men and for women >21
    	4. For govt jobs: (min:18, max:32)  for men and (min:18, max:34) for 	women
   	 	5. For driving licence: (min:18, max:60) for men and women
	Eligibility:
   		1. theatre
			2.  Voting system
			3.  Marriage in india
			4.  For govt obs
			5. For driving licence:
	Enter an option:
		Gender:
			1. men
			2. women
	Enter an option:
		Enter an age of person:
   	


In [ ]:
print("1.Men\n2.Women")
 
gender = input("enter your gender")
age = input("enter an age:")
age = float(age)
if age>=18:
   print("He can apply for voting")
if 18<= age >= 60:
   print("He can apply for driving licence")
if gender == "1":
     if age>=5:
        print("he can got to movie")
     if age>=23:
        print("He can get married")
     if age>=18 and age<=34:
       print("He can apply for govt gobs")
elif  gender == "2": 
    if age>=7:
         print("she can got to movie")
    if age >= 18:
        print("sHe can apply for voting")
    if age>=23:
        print("sHe can get married")
    if age>=18 and age<=34:
        print("sHe can apply for govt gobs")


#### 18.show the menu:
   		 1. kids
    		2. Men's
   		 3. Women's
    		Show the corresponding message based on the selection.
		Option:1: you are a kid
                                           2: you are a gentlemen
                                            3: you are a good women

 Mention error message if the option value >3.


In [ ]:
menu=int(input("Enter 1 for kids\n 2 for mens\n 3 for women\n:"))
if menu==1:
    print("you are a kid")
if menu==2:
    print("you are a gentleman")
if menu==3:
    print("you are a good women")
if menu>3:
    print("Error Message")

#### 19.write a program to chcek given substring is there in actual string or not?
example: act="python is a pure object oriented programing language"
check whether “pure” is there in act or not.
Note: Use in operator


In [ ]:
act = input("python is a pure object oriented programming language:")
substring =input("pure")
if substring in act:
    print("substring is present in actual string:")
else:
    print("substring is not present in actual string:")

#### 20.Take three numbers from the user and decide which is big

In [ ]:
num1=10
num2=20
num3=30
if num1 > num2 and num1 > num3:
    print("The largest number is: {num1}")
elif num2 > num1 and num2 > num3:
    print("The largest number is: {num2}")
else:
    print("The largest number is: {num3}")

#### 21.Take age and gender from the user and decide whether he is eligible for 	marriage in India or not.
Age criteria: men age>24, women>21


In [ ]:
age = int(input("Enter your age:"))
gender = input("Enter your gender (men/women):")
if gender == "men" and age > 24:
    print("You are eligible for marriage in India:")
elif gender == "women" and age > 21:
    print("You are eligible for marriage in India:")
else:
    print("You are not eligible for marriage in India:")
        

#### 23.operating systems:
	1.windows
	2.android
	3.mac
Enter an option:

If the user enters 1 then show "Goto first floor and buy windows laptop or mobile"
If the user enters 2 then show "Goto second floor and buy adroid mobiles"
If the user enters 3 then show "Goto third floor and buy mac laptop or iphones"
If the user enters other than 1 or 2 or 3 then show "There is only three floors, please select 1 or 2 or 3"


In [ ]:
print("Operating systems:")
print("1. Windows")
print("2. Android")
print("3. Mac")
option = int(input("Enter an option: "))
if option == 1:
    print("Goto first floor and buy windows laptop or mobile")
elif option == 2:
    print("Goto second floor and buy android mobiles")
elif option == 3:
    print("Goto third floor and buy mac laptop or iPhones")
else:
    print("There is only three floors, please select 1 or 2 or 3")


#### 24.Given an age, figure out whether someone's a baby, toddler, child, teenager, adult or old codger.

In [ ]:
age = int(input("Enter your age: "))
if age < 2:
    print("You are a baby.")
elif age >= 2 and age < 4:
    print("You are a toddler.")
elif age >= 4 and age < 13:
    print("You are a child.")
elif age >= 13 and age < 20:
    print("You are a teenager.")
elif age >= 20 and age < 60:
    print("You are an adult.")
else:
    print("You are an old codger.")

#### 25.Take two number a,b from the user and check whether a is divisible by b or not

In [ ]:
a = float(input("Enter the first number (a): "))
b = float(input("Enter the second number (b): "))
if b == 0:
    print("Division by zero is not allowed.")
elif a % b == 0:
    print("{a} is divisible by {b}.")
else:
    print("{a} is not divisible by {b}.")

#### 26.Take a letter from the user and print that letter belongs to which category I.e is it a small letter or capital letter or number or special symbol

In [ ]:
# Taking a single character input from the user
char = input("Enter a single character: ")
# Checking if the input is a single character
if len(char) != 1:
    print("Please enter only one character:")
else:
    ascii_value = ord(char)
if 65 <= ascii_value <= 90:
    print("The character is a capital letter:")
elif 97 <= ascii_value <= 122:
    print("The character is a small letter:")
elif 48 <= ascii_value <= 57:
    print("The character is a number:")
else:
    print("The character is a special symbol:")


#### 27.Take a number from the user and check whether it is prime?

In [ ]:
num = int(input("Enter the number to be checked: "))
if num > 1:
    for i in range(2, int(num/2)+1):
        if (num % i) == 0:
            print(num, "is not a prime number")
            break
        else:
            print(num, "is a prime number")
else:
    print(num, "is not a prime number")


#### 28.Take a string from the user and check contains only digits or not?

In [ ]:
input_str = input("Enter a string:")
if input_str.isdigit():
    print("The string contains only digits:")
else:
    print("The string does not contains only digits:")

 #### 29.Take a string from the user and check contains only  alphabets or not?

In [ ]:
input_str = input("Enter an alphabets:")
if input_str.isalpha():
    print("The string contains only alphabets:")
else:
    print("The string does not contains alphabets:")

#### 30.Take a string from the user and check contains only  special chars or not?

In [ ]:
string = input("Enter a string: ")
 
for char in string:
    if ('a' <= char <= 'z') or ('A' <= char <= 'Z') or ('0' <= char <= '9'):
        print("The string contains non-special characters.")
        break
else:
    print("The string contains only special characters.")

    

#### 31.Take a string from the user and check contains only  capital letters or not?

In [ ]:
string = input("Enter a string: ")

for char in string:
    if not ('A' <= char <= 'Z'):
        print("The string contains non-capital letters.")
        break
else:
    print("The string contains only capital letters.")

#### 32.Take a string from the user and check contains only  small letters or not?

In [ ]:
string = input("Enter a string: ")

for char in string:
    if not ('a' <= char <= 'z'):
        print("The string  does not contains small letters.")
        break
else:
    print("The string contains only small letters.")

#### 33. WAP to replace last n occurrence of give string.
For example:”apple,orange,apple,grape,orange,apple,apple,orange”
source: “apple”
last occurrences: 2
replace with: APPLE
output:”apple,orange,apple,grape,orange,APPLE,APPLE,orange”



In [ ]:
s = "apple,orange,apple,grape,orange,apple,apple,orange"
"APPLE".join(s.rsplit("apple",2))

#### 34.WAP to check given string contains numbers or not. it should consider float numbers also?

In [ ]:
user_input = input("Enter a string it contains numbers and float values:")
for i in user_input:
    if(i >= "0" and i <= "9"):
        print("contains numbers")
        break
    else:
        print("not contains numbers")

#### 35.Convert the total string in to lower case. Without using lower() function?

In [1]:
user_input = input("Enter a word with alphabets : ")
user_input = list(user_input)
for i in range(0, len(user_input)):
    if ((user_input[i] >= "A" and user_input[i] <= "Z")) :
        user_input[i] = chr(ord(user_input[i]) + 32)

print("".join(user_input))

Enter a word with alphabets :  ABCabc


abcabc


#### 36.Convert the total string in to upper case. Without using upper() function?

In [2]:
user_input = input("Enter a word with alphabets:")
user_input = list(user_input)
for i in range(0, len(user_input)):
    if ((user_input[i] >= "a" and user_input[i] <= "z")):
        user_input[i] = chr(ord(user_input[i]) - 32)
print("" .join(user_input))

Enter a word with alphabets: ABCabc


ABCABC


 #### 37.Show the below menu to the user until and until user select quit and display corresponding os message
	
'''
Menu:
1. windows
2. Linux
3. Mac
4. quit


In [3]:
condition = True
while condition:
    a = "Menu: \n1. windows \n2. Linux \n3. Mac \n4. quit"
    print(a)
    user_input = input("Select a number from above options : ")
    if "1" == user_input:
        print(" You selected Windows ")
        condition = True
    elif "2" == user_input:
        print("You selected Linux")
        condition = True
    elif "3" == user_input:
        print("You selected Mac")
        condition = True
    elif "4" == user_input:
        print("Thank You!")
        condition = False
    else:
        print("===>  Select from given options please  <===")
        condition = True

Menu: 
1. windows 
2. Linux 
3. Mac 
4. quit


Select a number from above options :  3


You selected Mac
Menu: 
1. windows 
2. Linux 
3. Mac 
4. quit


Select a number from above options :  2


You selected Linux
Menu: 
1. windows 
2. Linux 
3. Mac 
4. quit


Select a number from above options :  1


 You selected Windows 
Menu: 
1. windows 
2. Linux 
3. Mac 
4. quit


Select a number from above options :  4


Thank You!


 #### 38.Take a string from the user and check contains at least one digit or not?

In [9]:
user_input = input("Enter a string it contains numbers:")
for i in user_input:
    if (i >= "0" and i <= "9"):
        print("contains numbers")
        break
    else:
        print("not contains numbers")

Enter a string it contains numbers: 12345


contains numbers


#### 39.Take a string from the user and check contains at least one alphabets or not?

In [11]:
user_input = input("Enter a string it contains alphabets:")
for i in user_input:
    if(i >= "A" and i <= "Z") or (i >= "a" and i <= "z"):
        print("contains alphabets")
        break
    else:
        print("Not contains alphabets")

Enter a string it contains alphabets: 1245A


Not contains alphabets
Not contains alphabets
Not contains alphabets
Not contains alphabets
contains alphabets


#### 40.Take a string from the user and check contains at least one chars or not?

In [13]:
user_input = input("Enter a string it contains char:")
for i in user_input:
    if not((i >= "A" and i <= "Z") or (i >= "a" and i <= "z") or (i >= "0" and i <= "9")):
        print("contains chars")
        break
    else:
        print("not contains chars")

Enter a string it contains char: sfgs@


not contains chars
not contains chars
not contains chars
not contains chars
contains chars


#### 41.Take a string from the user and check contains at least one capital letter or not?

In [14]:
user_input = input("Enter a string it contains capital letters:")
for i in user_input:
    if(i >= "A" and i <= "Z"):
        print("contains capital letters")
        break
    else:
        print("not contains capital letters")

Enter a string it contains capital letters: abcdegH


not contains capital letters
not contains capital letters
not contains capital letters
not contains capital letters
not contains capital letters
not contains capital letters
contains capital letters


 #### 42 Take a string from the user and check contains at least one small letter or not?

In [16]:
user_input = input("Enter a string it contains small letters:")
for i in user_input:
    if(i >= "a" and i <= "z"):
        print("contains small letters")
        break
    else:
        print("not contains small letters")

Enter a string it contains small letters: ABCDEF


not contains small letters
not contains small letters
not contains small letters
not contains small letters
not contains small letters
not contains small letters


#### 43.Print the first 100 odd numbers?

In [19]:
for i in range(1, 200):
    if i % 2 != 0:
        print(i, end=",")


1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95,97,99,101,103,105,107,109,111,113,115,117,119,121,123,125,127,129,131,133,135,137,139,141,143,145,147,149,151,153,155,157,159,161,163,165,167,169,171,173,175,177,179,181,183,185,187,189,191,193,195,197,199,

#### 44.Determine the factors of a number entered  by the user?

In [20]:
user_input = input("enter a number : ")
user_input = int(user_input)

for i in range(1, user_input+1):
    if user_input % i == 0:
        print(i, end= " ")


enter a number :  40


1 2 4 5 8 10 20 40 

#### 45.Play a number guessing game (User enters a guess, you print YES or Higher or Lower). This should continue until and until user gives a correct number or want to quit in the middle.
Get a hidden number by using random.randint(1,100)


In [27]:
import random
number = random.randint(1, 100)
# print(number)
condition = True
while condition:
    user_input = input("enter a number for guess : ")
    user_input = int(user_input)
    if number == user_input:
        print("YES")
        condition = False
    elif number > user_input:
        print("Lower")
        condition = True
    else:
        print("Higher")
        condition = True

enter a number for guess :  90


Lower


enter a number for guess :  10


Lower


enter a number for guess :  50


Lower


enter a number for guess :  85


Lower


enter a number for guess :  88


Lower


enter a number for guess :  86


Lower


enter a number for guess :  87


Lower


enter a number for guess :  89


Lower


enter a number for guess :  86


Lower


enter a number for guess :  88


Lower


enter a number for guess :  90


Lower


enter a number for guess :  75


Lower


enter a number for guess :  76


Lower


enter a number for guess :  77


Lower


enter a number for guess :  60


Lower


enter a number for guess :  55


Lower


enter a number for guess :  45


Lower


enter a number for guess :  78


Lower


enter a number for guess :  70


Lower


enter a number for guess :  55


Lower


enter a number for guess :  78


Lower


enter a number for guess :  95


YES


#### 45.Take two numbers from the user a,b check whether a is divisible by b or not?

In [28]:
a = input("Enter a number")
a = int(a)
b = input("Enter a number")
b = int(b)
print(a % b)
if a % b == 0:
    print("a is divisible by b")
else:
    print(" a is not divisible by b")

Enter a number 14
Enter a number 2


0
a is divisible by b


 #### 46.Find the sum of all the multiples of 3 or 5 below 1000?

In [53]:
sum_of_3 = 0
sum_of_5 = 0
for i in range(1, 1000):
    if i % 3 == 0:
        sum_of_3 += i
    if i % 5 == 0:
        sum_of_5 += i
print("sum of multiples of 3 is", sum_of_3)
print("sum of multiples of 5 is", sum_of_5)
         


sum of multiples of 3 is 166833
sum of multiples of 5 is 99500


#### 47.Write a program to find out big of two numbers?

In [30]:
user_input = input("Enter a number:")
user_input2 = input("Enter a number:")
user_input = float(user_input)
user_input2 = float(user_input2)
if user_input > user_input2:
    print("Big number is", int(user_input))
else:
    print("Big number is", int(user_input2))
    
                   

Enter a number: 20
Enter a number: 30


Big number is 30


#### 48.Write a program to find out biggest number in the given numbers?

In [31]:
user_input = input("Enter a Number :")
user_input = list(user_input)
big = int(user_input[0])
for i in user_input:
    if int(i) >= big:
        big = int(i)
print(big)


Enter a Number : 125321


5


#### 49.find out the index of  third occurrence of given substring?

In [34]:
user_input = input("Enter a Number:")
user_input = list(user_input)
big = int(user_input[0])
for i in user_input:
    if int(i) >= big:
        big = int(i)
print(big)

Enter a Number: 12456


6


#### 50.find out the index nth occurrence of given substring?

In [36]:
user_input = input("Enter a String ")
user_input_2 = input("Enter a sub string")
count = 0
length = 0
list_words = user_input.split(" ")
for i in range(0, len(list_words)):
    if list_words[i] == user_input_2:
        count += 1
        if count == 3:
            break
        length += len(list_words[i])
    else:
        length += len(list_words[i])
user_input = " ".join(list_words)
if count == 3:
    print(length+count)

Enter a String  teja p 2 teja 2 teja
Enter a sub string teja


14


#### 51.find out the index nth occurrence of given substring?

In [37]:
user_input = input("Enter a String ")
user_input_2 = input("Enter a sub string")
count = 0
length = 0
list_words = user_input.split(" ")
for i in range(0, len(list_words)):
    if list_words[i] == user_input_2:
        count += 1
        length += len(list_words[i])
    else:
        length += len(list_words[i])
user_input = " ".join(list_words)

print(length+count)

Enter a String  teju teju teju
Enter a sub string teju


15


#### 51.Take some single digit numbers from the user and findout min, maximum, sum, average?

In [39]:
n = input("enter a number for no of inputs : ")
n = int(n)
list_a = []
for i in range(n):
    user_input = input(" enter a number ")
    list_a.append(int(user_input))
min_of = list_a[0]
max_of = list_a[0]
sum_of = 0
for i in list_a:
    sum_of += i
    if min_of >= i:
        min_of = i
    if max_of  <= i:
        max_of = i
print("mimimunm of a given numbers ", min_of)
print("Maximum of a given numbers ", max_of)
print("sum of given numbers ", sum_of)
print("Average of given numbers", (sum_of / n))

enter a number for no of inputs :  3
 enter a number  30
 enter a number  40
 enter a number  20


mimimunm of a given numbers  20
Maximum of a given numbers  40
sum of given numbers  90
Average of given numbers 30.0


 #### 52.print the number in proper mathematical way?
	Consider that we have 6 digit numbers.
Number format  WAP> 10 -> 000010
       		100 ->  000100
      		1000 ->  001000
 		 2345678  ->  2345678
	If the number has more than 6 digits then print as it is.


In [40]:
user_input = input("enter a number ")
length = len(user_input)
if length < 7:
    print("0" * (6- length) + user_input) 

enter a number  321


000321


 #### 53.names  ="emp1,emp2,emp3,emp4" iterate through the employee names?

In [41]:
names ="emp1,emp2,emp3,emp4"
for i in names.split(","):
    print(i, input("enter emp name " ))

enter emp name  123


emp1 123


enter emp name  456


emp2 456


enter emp name  79


emp3 79


enter emp name  125


emp4 125


 #### 54.Take actual string, source string, destination string. replce first nth occurrences of source string with destination string of actual string?

In [42]:
user_input = input("Enter a sentence ")
list_a = user_input.split()
print(list_a)
source = input("enter a sub string")
n = input("enter a occurances of numbers ")
n = int(n)
destination = input("enter a string to change")
count = 0
for i in range(0, len(list_a)):
    print(list_a[i])
    if list_a[i] == source :
        list_a[i] = destination
        count += 1
        if count == n:
            break
print(" ".join(list_a))

Enter a sentence  teju p teju p teju p


['teju', 'p', 'teju', 'p', 'teju', 'p']


enter a sub string p
enter a occurances of numbers  2
enter a string to change pilly


teju
p
teju
p
teju pilly teju pilly teju p


 #### 55. Take a two numbers from the user and do below menu driven operations
 1. addition
 2. multiples
 3.division
 4.sqrt
 5. pow    a**b
 6.subtraction
 After selection do the corresponding operation.
 Note: user may give int, or float numbers. You should check whether it is proper digits    or not. 
 I.e the user given string should be in the position to convert to float. 
 Other wise show the “inproper string given” Error.


In [44]:
user_input = input("enter a number :")
user_input = float(user_input)
user_input_1 = input("enter a number :")
user_input_1 = float(user_input_1)

print("1. addition \n2. multiples \n3.division \n4.sqrt \n5. pow a**b \n6.subtraction")
select = input("select from above options : ")
if select == "1":
    print("addition of :", user_input + user_input_1)
elif select == "2":
    print("multiples  of :", user_input * user_input_1)
elif select == "3":
    print("division   of :", user_input / user_input_1)
elif select == "4":
    print("sqrt of :", user_input **2,  user_input_1**2 )
elif select == "5":
    print("pow a**b  of :", user_input ** user_input_1 )
elif select == "6":
    print("subtraction  of :", user_input - user_input_1)
else:
    print("selected inproper option")
    



enter a number : 3
enter a number : 4


1. addition 
2. multiples 
3.division 
4.sqrt 
5. pow a**b 
6.subtraction


select from above options :  2


multiples  of : 12.0


 #### 56.Take numbers from the user and find out min, maximum, sum, average?

In [45]:

n = input("enter a number for no of inputs : ")
n = int(n)
list_a = []
for i in range(n):
    user_input = input(" enter a number ")
    list_a.append(int(user_input))
min_of = list_a[0]
max_of = list_a[0]
sum_of = 0
for i in list_a:
    sum_of += i
    if min_of >= i:
        min_of = i
    if max_of  <= i:
        max_of = i
print("mimimunm of a given numbers ", min_of)
print("Maximum of a given numbers ", max_of)
print("sum of given numbers ", sum_of)
print("Average of given numbers", (sum_of / n))

enter a number for no of inputs :  4
 enter a number  6
 enter a number  9
 enter a number  8
 enter a number  5


mimimunm of a given numbers  5
Maximum of a given numbers  9
sum of given numbers  28
Average of given numbers 7.0


#### 57. l=[1,2,3,5,7,8,9,10,11,12,13,20,22,23,24,25,26,27,20,21,22,4] find out how many even numbers are there and how many odd numbers are there and how many positive numbers are there and how many negative numbers are there and how many prime numbers are there and how many perfect numbers are there and how many Armstrong numbers are there and how many palindrome numbers are there.

In [47]:
ve_no = 0
ne_no = 0
even_no = 0
odd_no = 0
prime_no = 0
perfect_no = 0
armstrong_no = 0
palindrome_no = 0

l=[1,2,3,5,7,8,9,10,11,12,13,20,22,23,24,25,26,27,20,21,22,4]
for i in l:
    if i >=0:
        ve_no += 1
    else:
        ne_no += 1
    if i % 2 == 0:
        even_no += 1
    else:
        odd_no += 1
    if i <= 1:
        prime_no += 0
    else:
        count = 0
        for j in range(2, i+1):
            if i % j == 0:
                count += 1
        if count == 1:
            prime_no += 1
    if i <= 1:
        perfect_no += 0
    else:
        sum = 1
        for j in range(2, i+1):
            if i % j == 0:
                sum += j
                if i // j != j:
                    sum += i // j
        if sum == i:
            perfect_no += 1
    num_str = str(i)
    num_digits = len(num_str)
    sum_of_powers = 0
    
    for j in num_str:
        sum_of_powers += (int(j) ** num_digits)

    if sum_of_powers == i:
        armstrong_no += 1
    if str(i) == str(i)[::-1]:
        palindrome_no += 1

print("Even numbers:", even_no)
print("Odd numbers:", odd_no)
print("Positive numbers:", ve_no)
print("Negative numbers:", ne_no)
print("Prime numbers:", prime_no)
print("Perfect numbers:", perfect_no)
print("Armstrong numbers:", armstrong_no)
print("Palindrome numbers:", palindrome_no)

Even numbers: 11
Odd numbers: 11
Positive numbers: 22
Negative numbers: 0
Prime numbers: 7
Perfect numbers: 0
Armstrong numbers: 8
Palindrome numbers: 11


#### 58. Take a string from the user and find out how many digits are there, how many special symbols are there, how many small letters are there, how many caps are there?

In [48]:
user_input = input("Enter a string : ")
digit_count = 0
symbol_count = 0
small_letter = 0
captial_count = 0
for i in user_input:
    if (i >= "a" and i <= "z"):
        small_letter += 1
    if (i >= "A" and i <= "Z"):
        captial_count += 1
    if (i >= "0" and i <= "9"):
        digit_count += 1
    else:
        symbol_count += 1
print("Digits:", digit_count)
print("Special symbols:", symbol_count)
print("Small letters:", small_letter)
print("Capital letters:", captial_count)


Enter a string :  Hello teja 123 @ aja %5vhjghjg


Digits: 4
Special symbols: 26
Small letters: 18
Capital letters: 1


#### 59. Take a char from the user and find out how many number of occurrences are there in given string?

In [50]:
user_input = input("Enter a string :")
user_input_1 = input("Enter a char :")
count = 0
for i in user_input:
    if i == user_input_1:
        count += 1

print("no of occurances are : ", count)

Enter a string : rtyfyfyuf hhiuhuih
Enter a char : h


no of occurances are :  4


 #### 60.Take a element from the user and find out how many times the  element occurred in given list?

In [51]:
list_a = [1,2,3,4,5,2,5,2,6,1,2,5,2,1,5,4,3,8,9,2]
user_input = input("Enter a number :")
user_input = int(user_input)
count = 0
for i in list_a:
    if i == user_input:
        count += 1
print("no of  occurances are ", count)

Enter a number : 5


no of  occurances are  4


#### 61.Take an element from the user and find out how many number of occurrences are there in given tuple?

In [52]:
list_a = (1,2,3,4,5,2,5,2,6,1,2,5,2,1,5,4,3,8,9,2)
user_input = input("Enter a number :")
user_input = int(user_input)
count = 0
for i in list_a:
    if i == user_input:
        count += 1
print("no of  occurances are ", count)

Enter a number : 3


no of  occurances are  2


#### 62.Reverse the string without effecting the special symbols. It involves three variations. Write code for three variations.
         .Input: abc123,#$45def6%$^789$%^, output: $%^987%$^6fed54,#$321cba
        3.Input: abc123,#$45def6%$^789$%^, output: 9876fe,#$d54321%$^cba$%^
	
      64.Inout: "123,#$456%$^789$%^", Output: 321,#$654%$^987$%^
		Only numbers has to reverse.

